### PR TITLE
feat: PayPal env routing based on settings

### LINKED ISSUE #3413: Checks required for sandbox/live Paypal API routing
**Describe the bug**
By default, PayPal is being routed to the live API. In a dev/staging env, it should be routed to the sandbox API. Checks need to be present to route the request to the respective PayPal API.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #3413 

#### Short description of what this resolves:
Environment is being forced to a live environment without any param specified. TO properly route the API hit to sanbox/live, settings value is being passed.

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi @CosmicCoder96 I've checked in this locally and can confirm that it hits the sanbox API for test mode and live API for live mode. Please do confirm this.

File: app/components/paypal-button.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default Component.extend({
 
       paypal.Button.render({
         commit: true,
-
+        env: this.settings.paypalMode,
         style: {
```
Comment (uds5501) [inline app/components/paypal-button.js]: spacing error.

File: app/components/paypal-button.js [hunk 2]
```diff
@@ -78,9 +78,8 @@ export default Component.extend({
         }
       };
       paypal.Button.render({
-        env    : 'sandbox',
+        env    : this.settings.paypalMode,
         commit : true,
-
         style: {
```
Comment (uds5501) [inline app/components/paypal-button.js]: spacing error

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 The paypal button requires the environment parameter to be `production` instead of `live`.
`env    : this.settings.paypalMode === 'sandbox' ? this.settings.paypalMode : 'production'`

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 https://developer.paypal.com/docs/archive/checkout/integrate/#replace-sandbox-credentials-with-live-credentials

**Pull request merged**